content/v2.0/introduction/getting-started.md

```diff
@@ -138,29 +162,1261 @@ Visit your application in the browser at [http://localhost:2300](http://localhos
+    # Define your app settings here, for example:
+    #
+    # setting :my_flag, default: false, constructor: Types::Params::Bool
+
+    setting :database_url, constructor: Types::String
+  end
+end
+```
+
+Our bookshelf application will now raise an invalid settings error when it boots, unless a `DATABASE_URL` environment variable is present.
+
+In development and test environments, Hanami uses the [dotenv gem](https://github.com/bkeepers/dotenv) to load environment variables from `.env` files.
+
+We can now create `.env` and `.env.test` files in order to set `database_url` appropriately in development and test environments:
+
+```shell
+# .env
+DATABASE_URL=postgres://postgres:postgres@localhost:5432/bookshelf_development
+```
+
+```shell
+# .env.test
+DATABASE_URL=postgres://postgres:postgres@localhost:5432/bookshelf_test
+```
+
+<p class="notice">
+  You might need to adjust these connection strings based on your local postgres configuration.
+</p>
 
-## Commands
+To confirm that the `database_url` setting is working as expected, you can run `bundle exec hanami console` to start a console, then call the `database_url` method on your application's settings object.
 
-Hanami ships with useful commands for things like starting a console, inspecting routes and generating code.
+```shell
+bundle exec hanami console
+```
 
-To list available commands, run:
+```ruby
+bookshelf[development]> Hanami.app["settings"].database_url
+=> "postgres://postgres:postgres@localhost:5432/bookshelf_development"
+```
+
+And in test:
 
 ```shell
-bundle exec hanami --help
+HANAMI_ENV=test bundle exec hanami console
+```
+
+```ruby
+bookshelf[test]> Hanami.app["settings"].database_url
+=> "postgres://postgres:postgres@localhost:5432/bookshelf_test"
+```
+
+To ensure the database is cleaned between tests, add the following to a `spec/support/database.rb` file:
+
+```ruby
+# spec/support/database_cleaner.rb
+
+# frozen_string_literal: true
+
+require "database_cleaner-sequel"
+
+Hanami.app.prepare(:persistence)
+DatabaseCleaner[:sequel, db: Hanami.app["persistence.db"]]
 
-Commands:
-  hanami console                              # App REPL
-  hanami generate [SUBCOMMAND]
-  hanami install
-  hanami middlewares                          # List all the registered middlewares
-  hanami routes                               # Inspect application
-  hanami server                               # Start Hanami server
-  hanami version
+RSpec.configure do |config|
+  config.before(:suite) do
+    DatabaseCleaner.strategy = :transaction
+    DatabaseCleaner.clean_with(:truncation)
+  end
+
+  config.around(:each, type: :database) do |example|
+    DatabaseCleaner.cleaning do
+      example.run
+    end
+  end
+end
+```
+
+And then append the following line to `spec/spec_helper.rb`:
+
+```ruby
+require_relative "support/database_cleaner"
+```
+
+Finally, enable rom-rb's rake tasks for database migrations by appending the following to the `Rakefile`:
```

Comment:
It might be good to add another callout box at the end of this section just reminding the user that all of these manual steps will not be required as of Hanami 2.1, which will be released in early 2023.

Reply:
Done.